pkg/apis/pmemcsi/v1alpha1/deployment_types.go

```diff
@@ -40,19 +40,25 @@ func (mode *DeviceMode) String() string {
 	return string(*mode)
 }
 
+// +kubebuilder:validation:Enum=lvm,direct
 const (
 	// DeviceModeLVM represents 'lvm' device manager
 	DeviceModeLVM DeviceMode = "lvm"
 	// DeviceModeDirect represents 'direct' device manager
 	DeviceModeDirect DeviceMode = "direct"
 )
 
+// NOTE(avalluri): Due to below errors we stop setting
+// few CRD schema fields by prefixing those lines a '-'.
+// Once the below issues go fixed replace those '-' with '+'
+// Fails setting default(+kubebuilder:default=value): https://github.com/kubernetes-sigs/controller-tools/issues/478
+// Fails setting min/max for integers: https://github.com/helm/helm/issues/5806
```

Comment:
Please add an issue in our issue tracker for this.

Reply:
https://github.com/intel/pmem-csi/issues/735